pkg/kubelet/prober/worker.go

```diff
@@ -345,6 +345,8 @@ func getPodLabelName(pod *v1.Pod) string {
 		podName = strings.Join(podNameSlice[:len(podNameSlice)-1], "-")
 		if label, ok := pod.GetLabels()[apps.DefaultDeploymentUniqueLabelKey]; ok {
 			podName = strings.ReplaceAll(podName, fmt.Sprintf("-%s", label), "")
+		} else if label, ok := pod.GetLabels()[apps.LegacyDeploymentUniqueLabelKey]; ok {
+			podName = strings.ReplaceAll(podName, fmt.Sprintf("-%s", label), "")
```

Comment:
nit: pod.GetLabels() result can be assigned to a local variable to avoid calling it twice.